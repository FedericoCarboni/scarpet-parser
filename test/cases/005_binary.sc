// Binary Operator precedence test
1 + (2 + 3) * 4 ^ 5 / 2 % 3
