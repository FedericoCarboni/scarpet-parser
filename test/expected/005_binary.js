/** @type {import('../../lib/Parser.js').Node} */
export const root = {
    kind: 'BinaryExpression',
    operator: '+',
    lvalue: {
        kind: 'NumberLiteral',
        literal: '1',
        value: 1n,
        start: {
            offset: 35,
            line: 1,
            character: 0,
        },
        end: {
            offset: 36,
            line: 1,
            character: 1,
        },
    },
    rvalue: {
        kind: 'BinaryExpression',
        operator: '%',
        lvalue: {
            kind: 'BinaryExpression',
            operator: '/',
            lvalue: {
                kind: 'BinaryExpression',
                operator: '*',
                lvalue: {
                    kind: 'ParenthesisedExpression',
                    value: {
                        kind: 'BinaryExpression',
                        operator: '+',
                        lvalue: {
                            kind: 'NumberLiteral',
                            literal: '2',
                            value: 2n,
                            start: {
                                offset: 40,
                                line: 1,
                                character: 5,
                            },
                            end: {
                                offset: 41,
                                line: 1,
                                character: 6,
                            },
                        },
                        rvalue: {
                            kind: 'NumberLiteral',
                            literal: '3',
                            value: 3n,
                            start: {
                                offset: 44,
                                line: 1,
                                character: 9,
                            },
                            end: {
                                offset: 45,
                                line: 1,
                                character: 10,
                            },
                        },
                        comment: '// Binary Operator precedence test\n',
                        start: {
                            offset: 40,
                            line: 1,
                            character: 5,
                        },
                        end: {
                            offset: 45,
                            line: 1,
                            character: 10,
                        },
                    },
                    start: {
                        offset: 39,
                        line: 1,
                        character: 4,
                    },
                    end: {
                        offset: 46,
                        line: 1,
                        character: 11,
                    },
                },
                rvalue: {
                    kind: 'BinaryExpression',
                    operator: '^',
                    lvalue: {
                        kind: 'NumberLiteral',
                        literal: '4',
                        value: 4n,
                        start: {
                            offset: 49,
                            line: 1,
                            character: 14,
                        },
                        end: {
                            offset: 50,
                            line: 1,
                            character: 15,
                        },
                    },
                    rvalue: {
                        kind: 'NumberLiteral',
                        literal: '5',
                        value: 5n,
                        start: {
                            offset: 53,
                            line: 1,
                            character: 18,
                        },
                        end: {
                            offset: 54,
                            line: 1,
                            character: 19,
                        },
                    },
                    comment: '// Binary Operator precedence test\n',
                    start: {
                        offset: 49,
                        line: 1,
                        character: 14,
                    },
                    end: {
                        offset: 54,
                        line: 1,
                        character: 19,
                    },
                },
                comment: '// Binary Operator precedence test\n',
                start: {
                    offset: 39,
                    line: 1,
                    character: 4,
                },
                end: {
                    offset: 54,
                    line: 1,
                    character: 19,
                },
            },
            rvalue: {
                kind: 'NumberLiteral',
                literal: '2',
                value: 2n,
                start: {
                    offset: 57,
                    line: 1,
                    character: 22,
                },
                end: {
                    offset: 58,
                    line: 1,
                    character: 23,
                },
            },
            comment: '// Binary Operator precedence test\n',
            start: {
                offset: 39,
                line: 1,
                character: 4,
            },
            end: {
                offset: 58,
                line: 1,
                character: 23,
            },
        },
        rvalue: {
            kind: 'NumberLiteral',
            literal: '3',
            value: 3n,
            start: {
                offset: 61,
                line: 1,
                character: 26,
            },
            end: {
                offset: 62,
                line: 1,
                character: 27,
            },
        },
        comment: '// Binary Operator precedence test\n',
        start: {
            offset: 39,
            line: 1,
            character: 4,
        },
        end: {
            offset: 62,
            line: 1,
            character: 27,
        },
    },
    comment: '// Binary Operator precedence test\n',
    start: {
        offset: 35,
        line: 1,
        character: 0,
    },
    end: {
        offset: 62,
        line: 1,
        character: 27,
    },
};
