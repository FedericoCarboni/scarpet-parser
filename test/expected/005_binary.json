{
  "errors": [],
  "warnings": [],
  "root": {
    "kind": "BinaryExpression",
    "operator": "+",
    "lvalue": {
      "kind": "NumberLiteral",
      "literal": "1",
      "value": "1",
      "start": {
        "pos": 35,
        "row": 1,
        "col": 0
      },
      "end": {
        "pos": 36,
        "row": 1,
        "col": 1
      }
    },
    "rvalue": {
      "kind": "BinaryExpression",
      "operator": "%",
      "lvalue": {
        "kind": "BinaryExpression",
        "operator": "/",
        "lvalue": {
          "kind": "BinaryExpression",
          "operator": "*",
          "lvalue": {
            "kind": "ParenthesisedExpression",
            "value": {
              "kind": "BinaryExpression",
              "operator": "+",
              "lvalue": {
                "kind": "NumberLiteral",
                "literal": "2",
                "value": "2",
                "start": {
                  "pos": 40,
                  "row": 1,
                  "col": 5
                },
                "end": {
                  "pos": 41,
                  "row": 1,
                  "col": 6
                }
              },
              "rvalue": {
                "kind": "NumberLiteral",
                "literal": "3",
                "value": "3",
                "start": {
                  "pos": 44,
                  "row": 1,
                  "col": 9
                },
                "end": {
                  "pos": 45,
                  "row": 1,
                  "col": 10
                }
              },
              "comment": "// Binary Operator precedence test\n",
              "start": {
                "pos": 40,
                "row": 1,
                "col": 5
              },
              "end": {
                "pos": 45,
                "row": 1,
                "col": 10
              }
            },
            "start": {
              "pos": 39,
              "row": 1,
              "col": 4
            },
            "end": {
              "pos": 46,
              "row": 1,
              "col": 11
            }
          },
          "rvalue": {
            "kind": "BinaryExpression",
            "operator": "^",
            "lvalue": {
              "kind": "NumberLiteral",
              "literal": "4",
              "value": "4",
              "start": {
                "pos": 49,
                "row": 1,
                "col": 14
              },
              "end": {
                "pos": 50,
                "row": 1,
                "col": 15
              }
            },
            "rvalue": {
              "kind": "NumberLiteral",
              "literal": "5",
              "value": "5",
              "start": {
                "pos": 53,
                "row": 1,
                "col": 18
              },
              "end": {
                "pos": 54,
                "row": 1,
                "col": 19
              }
            },
            "comment": "// Binary Operator precedence test\n",
            "start": {
              "pos": 49,
              "row": 1,
              "col": 14
            },
            "end": {
              "pos": 54,
              "row": 1,
              "col": 19
            }
          },
          "comment": "// Binary Operator precedence test\n",
          "start": {
            "pos": 39,
            "row": 1,
            "col": 4
          },
          "end": {
            "pos": 54,
            "row": 1,
            "col": 19
          }
        },
        "rvalue": {
          "kind": "NumberLiteral",
          "literal": "2",
          "value": "2",
          "start": {
            "pos": 57,
            "row": 1,
            "col": 22
          },
          "end": {
            "pos": 58,
            "row": 1,
            "col": 23
          }
        },
        "comment": "// Binary Operator precedence test\n",
        "start": {
          "pos": 39,
          "row": 1,
          "col": 4
        },
        "end": {
          "pos": 58,
          "row": 1,
          "col": 23
        }
      },
      "rvalue": {
        "kind": "NumberLiteral",
        "literal": "3",
        "value": "3",
        "start": {
          "pos": 61,
          "row": 1,
          "col": 26
        },
        "end": {
          "pos": 62,
          "row": 1,
          "col": 27
        }
      },
      "comment": "// Binary Operator precedence test\n",
      "start": {
        "pos": 39,
        "row": 1,
        "col": 4
      },
      "end": {
        "pos": 62,
        "row": 1,
        "col": 27
      }
    },
    "comment": "// Binary Operator precedence test\n",
    "start": {
      "pos": 35,
      "row": 1,
      "col": 0
    },
    "end": {
      "pos": 62,
      "row": 1,
      "col": 27
    }
  }
}